src/core/instance/proxy.js

```diff
@@ -45,9 +55,11 @@ if (process.env.NODE_ENV !== 'production') {
   const hasHandler = {
     has (target, key) {
       const has = key in target
-      const isAllowed = allowedGlobals(key) || (typeof key === 'string' && key.charAt(0) === '_')
+      const isAllowed = allowedGlobals(key) ||
+        (typeof key === 'string' && key.charAt(0) === '_' && !(key in target.$data))
       if (!has && !isAllowed) {
-        warnNonPresent(target, key)
+        if (key in target.$data) warnStartingWithDollar(target, key)
```

Comment:
Does it have to be some property starting with preserved prefixes here? It feels a little weird to me that after these checks we concluded that the property starts with `_` or `$`.

Reply:
Proxying properties starting with $_ would be an option